### PR TITLE
Refactored search

### LINKED ISSUE #2455: [Search] Search does not work when we enter two words
Environment: Windows 10 browser: all
Reproducible: always
Build found: 23.03.2021

Preconditions
The GreenCity application is opened

Steps to reproduce

1. Choose two words that are exactly in the content (e.g. '20' and 'day')
2. Input into 'search' field '20 day'
3. Pay attention to the result

Actual result:
Message '0 items found' appears

Expected result:
First results where those two words appear together, but not necessarily one near another, the later results will contain at least one of those words

User story and test case links
User story #[545](https://github.com/ita-social-projects/GreenCity/issues/545)

### PR DESCRIPTION
**Pull request by VadymMakitra**

### CODE REVIEW

**Review by RossSihovsk — APPROVED** [review]

**Review by Natalia1910 — APPROVED** [review]

**Review by Max877866 — APPROVED** [review]

**Review by Zingerd — APPROVED** [review]

File: dao/src/main/java/greencity/repository/EcoNewsSearchRepo.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+    private long getEcoNewsCount(Predicate predicate) {
+        CriteriaQuery<Long> countQuery = criteriaBuilder.createQuery(Long.class);
+        Root<EcoNews> countEcoNewsRoot = countQuery.from(EcoNews.class);
+        countQuery.select(criteriaBuilder.count(countEcoNewsRoot)).where(predicate);
+        return entityManager.createQuery(countQuery).getSingleResult();
+    }
+
+    private List<Predicate> formEcoNewsLikePredicate(String searcingText, Root<EcoNews> root) {
+        Expression<String> title = root.get("title").as(String.class);
```
Comment (thesvider) [inline dao/src/main/java/greencity/repository/EcoNewsSearchRepo.java:62]: annotate your entity with @ FieldNameConstants

File: dao/src/main/java/greencity/repository/EcoNewsSearchRepo.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        Expression<String> text = root.get("text").as(String.class);
+        Expression<String> shortInfo = root.get("shortInfo").as(String.class);
+
+        List<Predicate> predicateList = new ArrayList<>();
+        Arrays.stream(searcingText.split(" ")).forEach(partOfSearchingText -> predicateList.add(
```
Comment (thesvider) [inline dao/src/main/java/greencity/repository/EcoNewsSearchRepo.java:67]: .map()
.collect(Collectors.toList())

File: dao/src/main/java/greencity/repository/EcoNewsSearchRepo.java [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+            criteriaBuilder.or(
+                criteriaBuilder.like(criteriaBuilder.lower(title), "%" + partOfSearchingText.toLowerCase() + "%"),
```
Comment (thesvider) [inline dao/src/main/java/greencity/repository/EcoNewsSearchRepo.java:69]: extract into private method.

File: service-api/src/main/java/greencity/service/EcoNewsService.java [hunk 4]
```diff
@@ -109,7 +109,7 @@
      * @param pageable    {@link Pageable}.
      * @param searchQuery query to search.
      * @return PageableDto of {@link SearchNewsDto} instances.
-     * @author Yurii Savchenko
+     * @author Vadym Maktra
```
Comment (thesvider) [inline service-api/src/main/java/greencity/service/EcoNewsService.java:112]: typo

**Review by thesvider — COMMENTED** [review]
resolve comments

**Pull request merged**